### PR TITLE
ASP.NET Core HTTPS development certificate support

### PR DESCRIPTION
**Pull request by javiercn**
Hey @livarcocc. I've put together a PR based on our conversation on Friday. There are a few small issues to figure out but I think the general approach is ok.

Added support for generating the HTTPS development certificate on the CLI first run experience.
There are a bunch of open issues:
* How/where to put the certificate generation component. (For this proof of concept I just inlined it, but we need to figure out the engineering to determine where to house the component and how to share it with the CLI).
* Add tests related to the functionality. (I'm just putting it all together to validate the experience).
* The implementation needs to live in dotnet, as it is netcoreapp2.0 (I didn't want to change the target framework of Microsoft.DotNet.Configurer nor cross-compile unless you tell me otherwise).
* Review messages shown to the user.

### CODE REVIEW

File: src/Microsoft.DotNet.Configurer/AspNetCertificateSentinel.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class AspNetCertificateSentinel : IAspNetCertificateSentinel
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Configurer/AspNetCertificateSentinel.cs]: Not for you, but at some point we should unify these Sentinel interfaces. I don't think really need multiple of these interfaces just to convey that they are a different file. But don't worry about it in this PR.

File: src/Microsoft.DotNet.Configurer/IAspNetCertificateSentinel.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public interface IAspNetCertificateSentinel : IDisposable
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Configurer/IAspNetCertificateSentinel.cs]: You don't need for it to implement IDisposable right?

File: src/Microsoft.DotNet.Configurer/DotnetFirstTimeUseConfigurer.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Configurer/DotnetFirstTimeUseConfigurer.cs]: Is this namespace really needed?

File: src/dotnet/AspNetCoreCertificateGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,223 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
```
Comment (livarcocc) [inline src/dotnet/AspNetCoreCertificateGenerator.cs]: Yep, this should not be here!

File: test/Microsoft.DotNet.Configurer.UnitTests/GivenADotnetFirstTimeUseConfigurer.cs [hunk 5]
```diff
@@ -18,6 +18,8 @@ public class GivenADotnetFirstTimeUseConfigurer
         private Mock<INuGetCachePrimer> _nugetCachePrimerMock;
```
Comment (livarcocc) [inline test/Microsoft.DotNet.Configurer.UnitTests/GivenADotnetFirstTimeUseConfigurer.cs]: We should add tests that cover the particular scenarios for this.

File: src/Microsoft.DotNet.Configurer/IAspNetCertificateSentinel.cs (hunk 2 above)
Comment (javiercn) [reply src/Microsoft.DotNet.Configurer/IAspNetCertificateSentinel.cs]: IFirstTimeUseNoticeSentinel and FirstTimeUseNoticeSentinel declared, so I just followed along, but neither this nor firstimeusenoticesentinel need them. I'll remove it from this one and I can remove it from the other one if you want me too.

**Comment by javiercn** [comment]
@livarcocc Updated

**Comment by javiercn** [comment]
@livarcocc Is there anything else do you want me to address here?

**Comment by livarcocc** [comment]
Just rebase to resolve conflicts and get the PR green.

**Comment by javiercn** [comment]
@livarcocc I think this is good to go. I'm not sure why it failed on Ubuntu and how to trigger the build again. All other builds are green

**Comment by javiercn** [comment]
@livarcocc Rebased this PR again to resolve new conflicts. Seems there are some issues with some of the building infrastructure.

**Comment by Eilon** [comment]
@livarcocc , right now @javiercn is working on the merge conflicts. Do you think once that's done we can get this merged? We want to avoid having to rebase more than necessary.

**Comment by livarcocc** [comment]
Works for me. As soon as it turns green we can merge it.

**Comment by javiercn** [comment]
Everything is green all checks passed

**Comment by javiercn** [comment]
Thanks!

**Pull request merged**